ports/stm/common-hal/pwmio/PWMOut.c

```diff
@@ -62,13 +63,10 @@ STATIC bool timer_get_optimal_divisors(uint32_t *period, uint32_t *prescaler,
 }
 
 void pwmout_reset(void) {
-    uint16_t never_reset_mask = 0x00;
     for (int i = 0; i < TIM_BANK_ARRAY_LEN; i++) {
         if (!never_reset_tim[i]) {
-            reserved_tim[i] = 0x00;
-            tim_frequencies[i] = 0x00;
-        } else {
-            never_reset_mask |= 1 << i;
+            tim_channels_taken[i] = 0x00;
```

Comment:
this doesn't actually modify any of the registers, so won't the pwm peripheral keep doing whatever it was doing before reset()?  Are we depending on some other way of stopping the physical output on the pin, like setting the pin's function to gpio or disabled? This might merit a comment.

Reply:
This is really another bug.

Reply:
On second thought, the timers are reset by `timers_reset()`, so I think this is OK.